lua/lean/ft.lua

```diff
@@ -45,6 +51,15 @@ function ft.detect(filename)
   vim.opt.filetype = filetype
 end
 
+--- Maybe make the current buffer `nomodifiable` unless configured not to.
+---
+--- Prevents unintentional modification of files in `_target` (i.e. Lean
+--- dependencies).
+function ft.__maybe_make_nomodifiable(fname)
+  if not options.make_deps_nomodifiable then return end
+  if fname:find('/_target/.*/.*.lean') then vim.bo.modifiable = false end
```

Comment:
Could we do the same thing for standard library files?

Reply:
Yes def good idea, moving this to draft since I had the same thought yesterday (I'm gonna change this to instead be a configurable set of patterns).